### DISCUSSION TITLE
Dnscrypt-proxy returns same cached result for requests from different edns subnet.

### DISCUSSION THREAD

**Discussion by ha-ku** [DNSCrypt/dnscrypt-proxy#2090] [Potential issues]
The behavior I expect is something like this: 

For a domain (for example, upos-sz-mirrorcoso1.bilivideo.com) taking advantage of edns information, it should resolved to different IP depending on the edns subnet sent to the server. For example, request A `dig @223.5.5.5 +subnet='13.52.132.0/24' upos-sz-mirrorcoso1.bilivideo.com` results in response A:
```
;; ANSWER SECTION:
upos-sz-mirrorcoso1.bilivideo.com. 1 IN CNAME   upos-sz-mirrorcoso1.bilivideo.com.cdn.dnsv1.com.
upos-sz-mirrorcoso1.bilivideo.com.cdn.dnsv1.com. 1 IN CNAME mirrorcoso1.acg-dk.sched.tdnsv2.com.
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     182.254.48.67
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     121.51.172.225
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     121.51.160.249
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     121.51.160.185
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     121.51.172.153
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     182.254.49.82
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     182.254.48.68
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     182.254.49.60
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     182.254.49.47
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     182.254.49.56
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     182.254.49.91
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     121.51.172.181
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     121.51.172.240
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     121.51.91.156
mirrorcoso1.acg-dk.sched.tdnsv2.com. 1 IN A     121.51.249.164
```
and request B `dig @223.5.5.5 +subnet='222.209.26.0/24' upos-sz-mirrorcoso1.bilivideo.com` results in response B:
```
;; ANSWER SECTION:
upos-sz-mirrorcoso1.bilivideo.com. 4 IN CNAME   upos-sz-mirrorcoso1.bilivideo.com.cdn.dnsv1.com.
upos-sz-mirrorcoso1.bilivideo.com.cdn.dnsv1.com. 4 IN CNAME upos-sz-mirrorcoso1.bilivideo.com.p7.qcloudcdn.com.
upos-sz-mirrorcoso1.bilivideo.com.p7.qcloudcdn.com. 4 IN A 113.219.152.116
upos-sz-mirrorcoso1.bilivideo.com.p7.qcloudcdn.com. 4 IN A 124.232.139.148
upos-sz-mirrorcoso1.bilivideo.com.p7.qcloudcdn.com. 4 IN A 124.232.139.149
```

When I do this test to the server I setup on 127.0.0.1 with dnscrypt-proxy, those different request mentioned above resulted in same response. If request A is sent first, both request A and request B result in response A. If request B is sent first, both request A and request B result in response B. 

Is this behavior intentional?

**Comment by lifenjoiner** [comment]
AFAIK, `dnscrypt-proxy` doesn't cache the response for different clients. I guess it is designed and used as a local server, rather than a public server.
If you really need correct responses to the tests above, you can try `cache = false` currently.
